### PR TITLE
Added CODING_CONVENTIONS.md

### PR DESCRIPTION
**Pull request by maribu**
#### What this PR does

1. Copied current content of https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions into `CODING_CONVENTIONS.md`
2. Applied the coding convention on the file
    - Removed white space errors (trailing newline, trailing spaces/tabs)
    - Changed indent level from 2 spaces to 4 spaces
    - Added line breaks to keep the file within the 80 chars per line limit
3. Updated the reference to the coding convention to refer to `CODING_CONVENTIONS.md`, rather than referring to the wiki.

#### What this PR did not do

- Not a single (non-whitespace-non-end-of-line)-character of the content of the coding convention was changed

#### Reasoning / Goals of this PR

- The coding conventions in the wiki could be changed without review / discussion
- People actually writing code will necessarily have access to the RIOT git repository, but might not check the Wiki

#### Non-Goals of this PR

- Discuss  or change any of the contents of the coding conventions. (This could be done in follow up PRs.)

### Testing procedure

- Compare the content of `CODING_CONVENTIONS.md` with the current (25th of October 8:47 UTC) state of the wiki

### Issues/PRs references

### CODE REVIEW

**Comment by miri64** [comment]
> #### Non-Goals of this PR
> * Discuss  or change any of the contents of the coding conventions. (This could be done in follow up PRs.)

(a long term goal of mine was to write at some point a more concise version, including the Linux stuff, as I think that document is too confrontational/Linus-Torvalds-Style)

**Comment by maribu** [comment]
I'd also like to move from C99 to C11. But I think we really should resist on proposing changes now and focus on whether it makes more sense to have it in the Repo instead of the Wiki.

**Comment by gschorcht** [comment]
Some time ago, we started the transition from the wiki to the online documentation with the goal to have the documentation in the repository and to have the documentation nearby the code. This has already been finished for some guides, the boards and the CPUs. Thus, the online documentation becomes more and more the unique location for all documentations.

I'm wondering if it would be possible (and makes sense) to change the doc generation in that way that all these files (`CODING_CONVENTIONS.md`, `CONTRIBUTING.md` also become part of the online documentation. Since doxygen supports markdown, it should theoretically be possible.

**Comment by miri64** [comment]
> Some time ago, we started the transition from the wiki to the online documentation with the goal to have the documentation in the repository and to have the documentation nearby the code. This has already been finished for some guides, the boards and the CPUs. Thus, the online documentation becomes more and more the unique location for all documentations.
> 
> I'm wondering if it would be possible (and makes sense) to change the doc generation in that way that all these files (`CODING_CONVENTIONS.md`, `CONTRIBUTING.md`, ...) also become part of the online documentation. Since doxygen supports markdown, it should theoretically be possible.

Definitely thinkable, but shouldn't be part of this PR

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -161,7 +161,7 @@ You can find more information about RIOT development procedure on this
 
 [existing-pull-requests]: https://github.com/RIOT-OS/RIOT/pulls
 [archived-pull-requests]: https://github.com/RIOT-OS/RIOT/pulls?q=is:pr+label:"State:+archived"
-[coding-conventions]: https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions
+[coding-conventions]: https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md
```
Comment (miri64) [inline CONTRIBUTING.md]:
```suggestion
[coding-conventions]: CODING_CONVENTIONS.md
```

should also work and makes it clearer for an off-line reader, that the document is in the checked-out tree.

File: MAINTAINING.md [hunk 2]
```diff
@@ -158,5 +158,5 @@ maintainers reasonable time to ACK before dismissing their review.
 [list of maintainers]: https://github.com/RIOT-OS/RIOT/wiki/Maintainers
 [Best Practices]: https://github.com/RIOT-OS/RIOT/wiki/Best-Practice-for-RIOT-Programming
 [Comparing build sizes]: https://github.com/RIOT-OS/RIOT/wiki/Comparing-build-sizes
-[Coding Conventions]: https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions
+[Coding Conventions]: https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md
```
Comment (miri64) [inline MAINTAINING.md]: Dito

**Comment by gschorcht** [comment]
> Definitely thinkable, but shouldn't be part of this PR

That's clear, it was just a question that I had in mind.

**Comment by maribu** [comment]
>     1. What about the [C++ coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-Conventions-Cplusplus)?

I'd place that to `CODING_CONVENTIONS_C++.md`. But I really want the discussion here to be focused on whether moving the code of conduct of the wiki is reasonable and postpone the discussions on the details. So I'd say that should be a separate PR.

**Comment by miri64** [comment]
Please squash!

**Comment by maribu** [comment]
OK. There seems to be no opposition to this, right?

**Comment by miri64** [comment]
Yepp, sorry for not merging it right away.

**Comment by miri64** [comment]
Updated [the wiki as well](https://github.com/RIOT-OS/RIOT/wiki/Home/8d2a6c8a0c744dcca6357e1a40d7bfeadf3ec31c)

**Pull request merged**